core/src/main/scala/org/apache/spark/sql/execution/ConnectionPool.scala

```diff
@@ -169,8 +171,8 @@ object ConnectionPool {
    * @return true if this was the last reference and entire pool was removed
    *         and false otherwise
    */
-  def removePoolReference(id: String): Boolean = {
-    val dsKey = idToPoolMap.remove(id)
+  def removePoolReference(id: String, userName: String): Boolean = {
+    val dsKey = idToPoolMap.remove((id, userName))
```

Comment:
This is not user specific rather needs to be done for all the users since the underlying table itself is being dropped. So iterate over the idToPoolMap and remove all entries with matching ids.